### PR TITLE
update to TileDB-2.15.0/TileDB-Java-0.17.2

### PR DESCRIPTION
**Pull request by DimitrisStaratzis**

### CODE REVIEW

File: src/main/java/io/trino/plugin/tiledb/TileDBClient.java [hunk 1]
```diff
@@ -283,13 +270,24 @@ public Config buildConfig()
...
+        Config tileDBConfig = new Config();
+        boolean updateCtx = false;
+
+        // Encryption
+        String currentKey = ctx.getConfig().get("sm.encryption_key");
+        boolean currentDifferent = !currentKey.equals(encryptionKey) && currentKey.length() > 0;
+        // add a key only if there is currently none or the current is different
+        if (encryptionType != null && encryptionKey != null
+                && (currentKey.equals("") || currentDifferent)) {
+            tileDBConfig.set("sm.encryption_type", encryptionType.toString().replace("TILEDB_", ""));
```
Comment (ihnorton) [inline src/main/java/io/trino/plugin/tiledb/TileDBClient.java:284]: What is this replace for? It looks kind of funky... Should we set up a mapping function instead?

File: src/main/java/io/trino/plugin/tiledb/TileDBClient.java [hunk 2]
```diff
@@ -283,13 +270,24 @@ public Config buildConfig()
...
-        Context localCtx = ctx;
+        Config tileDBConfig = new Config();
+        boolean updateCtx = false;
+
+        // Encryption
+        String currentKey = ctx.getConfig().get("sm.encryption_key");
+        boolean currentDifferent = !currentKey.equals(encryptionKey) && currentKey.length() > 0;
+        // add a key only if there is currently none or the current is different
+        if (encryptionType != null && encryptionKey != null
+                && (currentKey.equals("") || currentDifferent)) {
```
Comment (ihnorton) [inline src/main/java/io/trino/plugin/tiledb/TileDBClient.java:283]: I think `currentKey.equals("")` might be redundant with `currentDifferent`?

**Review by ihnorton — APPROVED** [review]
Minor nits, LGTM after addressing. Thanks.

File: src/main/java/io/trino/plugin/tiledb/TileDBClient.java (hunk 1 above)
Comment (DimitrisStaratzis) [reply src/main/java/io/trino/plugin/tiledb/TileDBClient.java:283]: TileDB-Java adds the ```TILEDB_``` prefix to all the different encryption types (currently only AES). It seems funky but should work every time.

File: src/main/java/io/trino/plugin/tiledb/TileDBClient.java (hunk 2 above)
Comment (DimitrisStaratzis) [reply src/main/java/io/trino/plugin/tiledb/TileDBClient.java]: oops!

**Review by ihnorton — APPROVED** [review]

**Pull request merged**